src/cpp/rtps/transport/UDPv4Transport.cpp

```diff
@@ -553,6 +554,49 @@ void UDPv4Transport::SetSocketOutboundInterface(
     getSocketPtr(socket)->set_option(ip::multicast::outbound_interface(asio::ip::address_v4::from_string(sIp)));
 }
 
+void UDPv4Transport::update_network_interfaces()
+{
+    for (auto& channelResources : mInputSockets)
+    {
+        for (UDPChannelResource* channelResource : channelResources.second)
+        {
+            if (channelResource->interface() == s_IPv4AddressAny)
+            {
+                std::vector<IPFinder::info_IP> locNames;
+                get_ipv4s_unique_interfaces(locNames, true);
+                for (const auto& infoIP : locNames)
+                {
+                    auto ip = asio::ip::address_v4::from_string(infoIP.name);
+                    try
+                    {
+                        channelResource->socket()->set_option(ip::multicast::join_group(
+                                    ip::address_v4::from_string(DEFAULT_METATRAFFIC_MULTICAST_ADDRESS), ip));
```

Comment:
What will happen when a multicast input channel different from the default one has been open, and `update_network_interfaces` is called?

Reply:
I do not think we should be worried by that specific case. This feature has been designed to work only when the locators are generated automatically by Fast DDS. Being that the case, if the user has opened a different multicast input channel (or several ones), the feature would not work because `RTPSParticipantImpl::internal_metatraffic_locators_` flag would not be set. I have move the call to this function in `RTPSParticipantImpl::update_attributes` so is called only when the default unicast metatraffic locators have changed.